src/edge.js

```diff
@@ -41,25 +47,46 @@ export async function handleErrors(request, func) {
     return new Response(err.stack, { status: 500 });
   }
 }
-// Admin APIs are forwarded to the durable object. They need the doc name as a query
-// parameter on the url.
+
+/**
+ * Admin APIs are forwarded to the durable object. They need the doc name as a query
+ * parameter on the url.
+ * @param {string} api
+ * @param {URL} url
+ * @param {Request} request
+ * @param {Env} env
+ */
 async function adminAPI(api, url, request, env) {
   const doc = url.searchParams.get('doc');
   if (!doc) {
     return new Response('Bad', { status: 400 });
   }
 
+  // check if shared token is configured and validate
+  if (env.COLLAB_SHARED_SECRET) {
+    if (request.headers.get('authorization') !== `token ${env.COLLAB_SHARED_SECRET}`) {
+      return new Response('Unauthorized', { status: 401 });
+    }
+  }
+
   const id = env.rooms.idFromName(doc);
 
   // eslint-disable-next-line no-console
   console.log('[worker] - Admin API', doc);
 
   const roomObject = env.rooms.get(id);
-
-  return roomObject.fetch(new URL(`${doc}?api=${api}`));
+  // TODO: check for roomObject === null ?
+  // note, that we cannot call DocRoom.handleApiCall directly w/o enabling
+  // RPC on on the durable objects.
+  const apiUrl = new URL(doc);
+  apiUrl.searchParams.set('api', api);
+  return roomObject.fetch(new Request(apiUrl));
```

Comment:
not sure why this worked. but the `fetch()` api expects a `Request` (tested on stage as well)